controllers/dspipeline_controller.go

```diff
@@ -294,6 +305,17 @@ func (r *DSPAReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.
 			return ctrl.Result{}, err
 		}
 
+		if dspa.Spec.APIServer.PipelineStorage == "kubernetes" && r.DeployWebhook {
```

Comment:
It seems like `DeployWebhook` doesn't get set back to true after the webhook is removed.

My suggestion is to always run `ReconcileWebhook` on every reconcile of a DSPA if dspa.Spec.APIServer.PipelineStorage == "kubernetes" and just remove `DeployWebhook`.